### ISSUE TITLE
[java] ArrayIsStoredDirectly false positive

### ISSUE THREAD

**Issue by kevin-wayne** [pmd/pmd#468]
**Rule Set:**
`rulesets/java/sunsecure.xml/ArrayIsStoredDirectly`

**Description:**
False positive for `ArrayIsStoredDirectly` when defensively copying an array (manually) and using the `this` keyword to refer to the instance variable. 

**Code Sample demonstrating the issue:**

```
public class Foo {
    private final int[] a;

    public Foo(int[] b) {
        this.a = new int[b.length];
        for (int i = 0; i < b.length; i++) {
            this.a[i] = b[i];  // triggers ArrayIsStoredDirectly
            a[i] = b[i];       // no warning
            this.a[i] = b[i];  // triggers ArrayIsStoredDirectly
        }
    }
}
```

**Running PMD through:** *CLI*
```
pmd-bin-5.7.0/bin/run.sh pmd -d . -rulesets java-sunsecure
Foo.java:4:	The user-supplied array 'b' is stored directly.
Foo.java:4:	The user-supplied array 'b' is stored directly.
```

**Comment by jsotuyod** [comment]
Definitely a bug. Thanks for the report!
